state-chain/pallets/cf-reputation/src/lib.rs

```diff
@@ -388,24 +388,27 @@ impl<T: Config> OffenceReporter for Pallet<T> {
 	type ValidatorId = T::ValidatorId;
 	type Offence = T::Offence;
 
-	fn report_many(offence: impl Into<Self::Offence>, validators: &[Self::ValidatorId]) {
+	fn report_many(
+		offence: impl Into<Self::Offence>,
+		validators: impl IntoIterator<Item = T::ValidatorId> + Clone,
```

Comment:
nit: this could have been split into a separate PR - not really relevant and adds lot more line changes to this one (which is already complex enough).